### PR TITLE
Retrieve/store status from CMAPI

### PR DESCRIPTION
**Pull request by patrickvinograd**
Only checks downstream if document has been submitted and is in a non-terminal status.

Not sure if error response from status call should throw BadGateway or just log and silently return the last recorded status. Open to suggestions.

### CODE REVIEW

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb [hunk 1]
```diff
@@ -25,5 +46,24 @@ def signed_url(guid)
       obj = s3.bucket(Settings.vba_documents.s3.bucket).object(guid)
       obj.presigned_url(:put, {})
     end
+
+    def status_in_flight?
+      IN_FLIGHT_STATUSES.include?(status)
+    end
+
+    def map_downstream_status(body)
+      response_object = JSON.parse(body)[0][0]
```
Comment (patrickvinograd) [inline modules/vba_documents/app/models/vba_documents/upload_submission.rb]: The PensionBurial::Service could be better encapsulated here.

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb [hunk 2]
```diff
@@ -3,13 +3,34 @@
 module VBADocuments
   class UploadSubmission < ActiveRecord::Base
     include SetGuid
+    include SentryLogging
+
+    IN_FLIGHT_STATUSES = %w[received processing].freeze
```
Comment (jkassemi) [inline modules/vba_documents/app/models/vba_documents/upload_submission.rb]: Is it possible that the record could move from an error state to a success, or would that always require a new submission?

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb [hunk 3]
```diff
@@ -3,13 +3,34 @@
...
     # every time object is serialized
     def get_location
       rewrite_url(signed_url(guid))
     end
 
+    def refresh_status!
+      if status_in_flight?
+        response = PensionBurial::Service.new.status(guid)
+        # TODO: remove
+        Rails.logger.info(response.body)
```
Comment (jkassemi) [inline modules/vba_documents/app/models/vba_documents/upload_submission.rb]: kill

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb [hunk 4]
```diff
@@ -25,5 +46,24 @@ def signed_url(guid)
...
+        self.status = 'received'
+      elsif response_object['status'] == 'In Process'
+        self.status = 'processing'
+      elsif response_object['status'] == 'Success'
+        self.status = 'success'
+      elsif response_object['status'] == 'Error'
+        self.status = 'error'
+        self.code = 'DOC202'
+        self.detail = "Downstream status: #{response_object['errorMessage']}"
+      end
```
Comment (jkassemi) [inline modules/vba_documents/app/models/vba_documents/upload_submission.rb]: All documented statuses accounted for; potentially worth tossing an exception if we're not able to map the status.

Comment (patrickvinograd) [reply modules/vba_documents/app/models/vba_documents/upload_submission.rb]: Done

File: modules/vba_documents/app/models/vba_documents/upload_submission.rb (hunk 2 above)
Comment (patrickvinograd) [reply modules/vba_documents/app/models/vba_documents/upload_submission.rb]: Inquiry out to CMAPI team

**Comment by jkassemi** [comment]
Looks good, aside from the lint failure. Adjustments can be made based on what central mail team responds with, but no reason to keep this from going out.

**Pull request merged**